File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/BaseActivity.java
package com.roman.kubik.spivanyklicejista.presentation;

import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;

/**
 * Base activity for all application activities. Binds {@link ButterKnife}
 * Created by kubik on 1/20/18.
 */

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        ButterKnife.bind(this);
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        ButterKnife.bind(this);
    }

    @Override
    public void setContentView(View view, ViewGroup.LayoutParams params) {
        super.setContentView(view, params);
        ButterKnife.bind(this);
    }
}

File: app/src/main/java/com/roman/kubik/songer/presentation/BaseActivity.kt
package com.roman.kubik.songer.presentation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import com.roman.kubik.songer.general.android.SpivanykApplication.Companion.component
import com.roman.kubik.songer.general.di.ActivityComponent
import com.roman.kubik.songer.general.di.ActivityModule

/**
 * Base activity for all application activities
 * Created by kubik on 1/20/18.
 */

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        injectActivity(component.getActivityComponent(ActivityModule(this)))

    }

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        ButterKnife.bind(this)
    }

    override fun setContentView(view: View?) {
        super.setContentView(view)
        ButterKnife.bind(this)
    }

    override fun setContentView(view: View?, params: ViewGroup.LayoutParams?) {
        super.setContentView(view, params)
        ButterKnife.bind(this)
    }

    abstract fun injectActivity(activityComponent: ActivityComponent)
}
